amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectorSettings.scala

```diff
@@ -80,44 +137,72 @@ object TemporaryQueueSourceSettings {
     TemporaryQueueSourceSettings(connectionProvider, exchange)
 }
 
-final case class AmqpReplyToSinkSettings(
-    connectionProvider: AmqpConnectionProvider,
-    failIfReplyToMissing: Boolean = true
+final class AmqpReplyToSinkSettings private (
+    val connectionProvider: AmqpConnectionProvider,
+    val failIfReplyToMissing: Boolean = true
 ) extends AmqpConnectorSettings {
   override final val declarations = Nil
+
+  def withFailIfReplyToMissing(failIfReplyToMissing: Boolean): AmqpReplyToSinkSettings =
+    copy(failIfReplyToMissing = failIfReplyToMissing)
+
+  private def copy(connectionProvider: AmqpConnectionProvider = connectionProvider,
+                   failIfReplyToMissing: Boolean = failIfReplyToMissing) =
+    new AmqpReplyToSinkSettings(connectionProvider, failIfReplyToMissing)
+
+  override def toString: String =
+    s"AmqpReplyToSinkSettings(connectionProvider=$connectionProvider, failIfReplyToMissing=$failIfReplyToMissing)"
 }
 
 object AmqpReplyToSinkSettings {
+  def apply(connectionProvider: AmqpConnectionProvider): AmqpReplyToSinkSettings =
+    new AmqpReplyToSinkSettings(connectionProvider)
 
   /**
    * Java API
    */
   def create(connectionProvider: AmqpConnectionProvider): AmqpReplyToSinkSettings =
     AmqpReplyToSinkSettings(connectionProvider)
+}
+
+final class AmqpSinkSettings private (
+    val connectionProvider: AmqpConnectionProvider,
+    val exchange: Option[String] = None,
+    val routingKey: Option[String] = None,
+    val declarations: immutable.Seq[Declaration] = Nil
+) extends AmqpConnectorSettings {
+
+  def withExchange(exchange: String): AmqpSinkSettings =
+    copy(exchange = Some(exchange))
+
+  def withRoutingKey(routingKey: String): AmqpSinkSettings =
+    copy(routingKey = Some(routingKey))
+
+  def withDeclaration(declaration: Declaration): AmqpSinkSettings =
+    copy(declarations = immutable.Seq(declaration))
```

Comment:
I think it should add to the current list of values and not drop things set before.

Reply:
Interesting. This would be different from all the other setters, but it would still have `with` prefix. I think that can catch users surprised.

We could have setters (adders) that do not drop current values, but then we should call them differently.

Reply:
Yes, could be surprising.
`addDeclaration` sounds OK, but let's skip that for now.